app/logic/Calendar/Event.ts

```diff
@@ -518,10 +538,15 @@ export class Event extends Observable {
   }
 
   async saveToServer(): Promise<void> {
-    if (!this.isIncomingMeeting && this.participants.hasItems && this.hasChanged()) {
-      this.calUID ??= crypto.randomUUID();
-      await this.outgoingInvitation.sendInvitations();
+    if (this.isIncomingMeeting || !this.unedited) {
+      return;
     }
+    this.calUID ??= crypto.randomUUID();
+    let removed = this.unedited.participants.subtract(this.participants);
+    // Use the original event when sending cancellations
+    this.unedited.outgoingInvitation.sendCancellations(removed);
+    let notify = this.hasChangedInvitation() ? this.participants : this.participants.subtract(this.unedited.participants);
+    await this.outgoingInvitation.sendInvitations(notify);
```

Comment:
Move this logic into `outgoingInvitation.sendCancellations()`, and keep the old code (including the old `if` block), only adapting the condition.